### ISSUE TITLE
Suggestion:  Adding new songs to light mode playlist not repositioning the window

### ISSUE THREAD

**Issue by sharmanr** [mpogue2/SquareDesk#1198]
I primarily use light mode and when adding songs to a playlist I find it most annoying that the  playlist is repositioned so that the new song (at the top or the bottom) is visible.  I then have to scroll all the way down to find the next song to add, and so on.
I would much prefer to keep where I am in the list of songs so I can add more.
The change to do this turned out (rather surprisingly!) simple to do.  For testing I created a boolean makeNewSongsVisible to control it:  I set it to false but for backward compatibility it could be set to true.   When moving a song to the top or bottom in light mode,  if the song is not already in the list then if makeNewSongsVisible is true it moves it (as it currently does) otherwise it doesn't.   In all other cases it acts as before.
Currently makeNewSongsVisible is for testing with the debugger;  it could be made in to a preference if the old behaviour was felt desirable.
What do you think?  Patch attached,

[playlist-new-song.txt](https://github.com/user-attachments/files/17403418/playlist-new-song.txt)

**Comment by mpogue2** [comment]
Sounds like a reasonable request!
I think the main users of Light Mode are now just you and @pkbott. 

So, the options are:
- A (the way it is today): always make the new item visible
- B: never make the new item visible (leave the view the way it was)
- C: make it a Preference to do A vs B

Paul, do you have an opinion on this?  I'm OK with it either way, although I generally would prefer that we take a position on A or B, rather than add a preference (C).  But, I'm actually OK with A, B, or C.

**Comment by pkbott** [comment]
I’m ok any way since I don’t use it much. Ready to try something FUN, UNIQUE and ***@***.*** Square Dance Group:  https://www.facebook.com/groups/squaredanceforfun/On Oct 17, 2024, at 1:38 AM, Mike Pogue ***@***.***> wrote:﻿
Sounds like a reasonable request!
I think the main users of Light Mode are now just you and @pkbott.
So, the options are:

A (the way it is today): always make the new item visible
B: never make the new item visible (leave the view the way it was)
C: make it a Preference to do A vs B

Paul, do you have an opinion on this?  I'm OK with it either way, although I generally would prefer that we take a position on A or B, rather than add a preference (C).  But, I'm actually OK with A, B, or C.

**Comment by mpogue2** [comment]
@sharmanr  I'm currently going through the older feature requests now (aka doing "backlog grooming").  

I think this Enhancement suggestion is obsolete, now that the Old Light Mode is gone.  The New Light mode (and New Dark mode) have separate palette slots to show playlists.  

But, if you think there's something else that needs to be done with the palette slots as far as visibility goes when an item is added, could you either modify this Issue, or open a new Issue for that feature?

Thanks!

**Comment by sharmanr** [comment]
Yes, it is no longer make sense, so please close or obsolete it.
